tcmur_device.c

```diff
@@ -207,26 +208,26 @@ void tcmu_notify_lock_lost(struct tcmu_device *dev)
 int tcmu_cancel_lock_thread(struct tcmu_device *dev)
 {
 	struct tcmur_device *rdev = tcmu_get_daemon_dev_private(dev);
-	void *join_retval;
 	int ret;
 
 	pthread_mutex_lock(&rdev->state_lock);
 	if (rdev->lock_state != TCMUR_DEV_LOCK_LOCKING) {
 		pthread_mutex_unlock(&rdev->state_lock);
 		return 0;
 	}
-	pthread_mutex_unlock(&rdev->state_lock);
 	/*
 	 * It looks like lock calls are not cancelable, so
 	 * we wait here to avoid crashes.
 	 */
 	tcmu_dev_dbg(dev, "Waiting on lock thread\n");
-	ret = pthread_join(rdev->lock_thread, &join_retval);
+
 	/*
 	 * We may be called from the lock thread to reopen the device.
 	 */
```

Comment:
Remove the comment above since that is not true any more.

Reply:
Has removed.